kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ingress/Ingress.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.ingress;
+
+import java.util.stream.Stream;
+
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.ServiceBuilder;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.proxy.config.VirtualClusterGateway;
+
+public interface Ingress {
```

Comment:
Not at all sure about this pattern. I've made it so we transform each KafkaProxyIngress into a stateless Ingress, which is the smart model class that knows what services are required, how many ports it needs etc. And then it's up to some external co-ordinator to allocate the ports.

Maybe there should be two bits, an `IngressFactory` which has `proxyPortsRequired()` and then a method to create an Ingress that takes the allocated port range.